lib/PuppeteerSharp.Tests/PublishSingleFileTests/PublishSingleFileTests.cs

```diff
@@ -0,0 +1,96 @@
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using PuppeteerSharp.Tests.Attributes;
+using Xunit;
+
+namespace PuppeteerSharp.Tests.SingleFileDeployment
+{
+    public class PublishSingleFileTests
+    {
+        [SkipNonWindowsFact]
+        public void ShouldWork()
+        {
+            var tempPath = Path.GetTempPath();
+            var actualFilePath = Path.Combine(tempPath, $"google.jpg");
+            var actualWindowsBinary = DotnetPublishSingleFile("PuppeteerSharp.Tests.SingleFileDeployment");
+
+            DeleteIfExists(actualFilePath);
+
+            using var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = actualWindowsBinary,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    WorkingDirectory = tempPath
+                }
+            };
+
+            process.Start();
+            var outputResult = GetStreamOutput(process.StandardOutput);
+            var errorResult = GetStreamOutput(process.StandardError);
+            var isExited = process.WaitForExit(80000);
+
+            if (!isExited)
+            {
+                process.Kill();
+            }
+
+            Assert.True(isExited);
+
+            Assert.True(File.Exists(actualFilePath), $"StdOut: {outputResult}\nStdErr: {errorResult}\n");
+        }
+
+        private static string GetStreamOutput(StreamReader stream)
+        {
+            var outputReadTask = Task.Run(() => stream.ReadToEnd());
+
+            return outputReadTask.Result;
+        }
+
+        private static string DotnetPublishSingleFile(string projectName)
+        {
+            var absolutePath = Path.GetFullPath(Path.Combine("../../../../", projectName));
+            var expectedBinaryPath = Path.Combine(absolutePath, $"publish/{projectName}.exe");
+
+            DeleteIfExists(expectedBinaryPath);
+
+            using var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = "dotnet",
+                    Arguments = $"publish {absolutePath} --configuration Release -r win-x64 -p:PublishSingleFile=true --self-contained false --use-current-runtime -o ./publish",
```

Comment:
If you remove `-r win-x64` then hopefully `--use-current-runtime` should pickup the current `RID`.

https://github.com/dotnet/sdk/issues/10449